gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -435,10 +454,17 @@ public void run() {
       try {
         GobblinServiceJobScheduler.this.runJob(this.jobConfig, this.jobListener);
         if (flowCatalog.isPresent() && removeSpec) {
+          while (!GobblinServiceJobScheduler.this.flowCatalog.get().exists(specUri)) {
```

Comment:
Should this block be replaced with a "wait with a timeout"? Is there a chance this thread may wait forever? If so, we should avoid that.